lib/ansible/template/__init__.py

```diff
@@ -359,11 +359,14 @@ def _is_unsafe(self, val):
         a key or value which contains jinja2 syntax and would otherwise
         lose the AnsibleUnsafe value.
         """
-        if isinstance(val, dict):
+        from ansible.vars.hostvars import HostVarsVars
+        if isinstance(val, (AnsibleUndefined, HostVarsVars)):
```

Comment:
Treating `HostVarsVars` as safe seems wrong. This check breaks my local test where it still templates the string returned by a module.

Reply:
yes, but flagging all of a host's vars as unsafe because one is unsafe is wrong, i was going to redo this approach as while it fixes the issue, it creates many more. But you already found better solution in your PR. Thanks!